core/shared/src/main/scala/scalaz/zio/Task.scala

```diff
@@ -0,0 +1,631 @@
+package scalaz.zio
+
+import scalaz.zio.Exit.Cause
+import scalaz.zio.internal.{ Executor, Platform }
+
+import scala.concurrent.ExecutionContext
+
+/*
+ * PHASE 1 DONE
+ * PHASE 2 DONE
+ * PHASE 3 DONE
+ * PHASE 4 DONE
+ * PHASE 6 DONE
+ */
+
+/*
+ * EXTENDS ZIO_E_Throwable
+ */
+object Task {
+
+  def apply[A](a: => A): Task[A] = effect(a)
+
+  type TaskE = Throwable
+  type TaskR = Any
+
+  /*
+   * ----------------------------------------------------------------------------------------------------------------------
+   * The Following is taken from ZIOFunctions.
+   * ----------------------------------------------------------------------------------------------------------------------
+   */
+  //  // ALL error types in this trait must be a subtype of `UpperE`.
+  //  type UpperE = Throwable
+  //  // ALL environment types in this trait must be a supertype of `LowerR`.
+  //  type LowerR = Any
+
+  /** Returns an effect that models failure with the specified error.
+   * The moral equivalent of `throw` for pure code.
+   * [[scalaz.zio.ZIO.fail]]
+   */
+  final def fail(error: TaskE): IO[TaskE, Nothing] = ZIO.fail(error)
```

Comment:
Replace all `IO[???, A]` in this file by `Task[A]`, and delete all the environmental functions (`provide`, `accessM`, `access`, `environment`, etc.), because `R = Any`, they won't be useful.